File: src/game_of_life/java_coderetreat_socramob/cr_socramob06/src/com/hemmerling/Cell.java
package com.hemmerling;

public class Cell {

	private boolean isAlive = true;
	private int numberOfNeighbours = 0;

	public Cell() {
		this.isAlive = true;
		this.numberOfNeighbours = 0;
	}

	public Cell(int numberOfNeighbours) {
		this.isAlive = true;
		if (numberOfNeighbours < 0) {
			throw new IllegalArgumentException(
					"Number of neighbos needs to be positive");
		}
	}

	public boolean isAlive() {
		this.isAlive = false;
		this.isAlive = !(this.numberOfNeighbours < 2);
		return this.isAlive;
	}
}

File: src/game_of_life/php_coderetreat_socramob/gameoflife1_phpunit/cell.php
<?php
/**
 * User: aachim
 * Date: 13.02.16
 * Time: 13:38
 */

class Cell {
    private $numberOfNeighbours;
    private $myLivingStatus = true;
    public function addNighbours($number) {
        $this->numberOfNeighbours = $number;
    }

    public function isAlive() {
        return (($this->numberOfNeighbours == 2) && $this->myLivingStatus ) || ($this->numberOfNeighbours == 3);
    }

    public function nextGen() {
        return $this;
    }

    public function livingStatus($status) {
        $this->myLivingStatus = $status;
    }
} 